recorder/src/main/scala/weco/concepts/recorder/MergeConcepts.scala

```diff
@@ -0,0 +1,56 @@
+package weco.concepts.recorder
+
+import grizzled.slf4j.Logging
+import weco.concepts.common.model.{AuthoritativeConcept, Concept, UsedConcept}
+
+object MergeConcepts extends Logging {
+  def apply(
+    authoritative: Option[AuthoritativeConcept],
+    used: Option[UsedConcept]
+  ): Concept = (authoritative, used) match {
+    case (Some(authoritative), Some(used)) =>
+      info(
+        s"Merging ${used.canonicalId} and ${authoritative.identifier.toString}"
+      )
+      merge(authoritative, used)
+    case (None, Some(used)) =>
+      info(s"Forwarding ${used.identifier.toString}")
+      fromUsedOnly(used)
+    case (Some(authoritative), None) =>
+      throw new IllegalArgumentException(
+        s"This error should never occur: we've been asked to merge a concept (${authoritative.identifier}) which isn't used in the catalogue"
+      )
+    case (None, None) =>
+      throw new IllegalArgumentException(
+        "This error should never occur: we've been asked to merge 2 concepts which don't exist"
+      )
```

Comment:
Is there a way to enforce the defined-ness of `used` in the type system?

Reply:
No, but your asking that question means I can improve this error message: `used` might not be defined if the recorder receives an ID for a used concept that doesn't exist, so in reality it means something has gone wrong in the aggregator (it thinks it's indexed something and has told the recorder about it, but for some reason it didn't actually do it)